debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/CaseSensitivenessIT.java

```diff
@@ -0,0 +1,148 @@
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.sqlserver;
+
+import java.sql.SQLException;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.fest.assertions.Assertions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.sqlserver.SqlServerConnectorConfig.SnapshotMode;
+import io.debezium.connector.sqlserver.util.TestHelper;
+import io.debezium.embedded.AbstractConnectorTest;
+import io.debezium.util.Testing;
+
+/**
+ * Integration test to verify behaviour of detabase with and without case sensistive names.
+ *
+ * @author Jiri Pechanec
+ */
+public class CaseSensitivenessIT extends AbstractConnectorTest {
+
+    private SqlServerConnection connection;
+
+    @Before
+    public void before() throws SQLException {
+        TestHelper.createTestDatabase();
+        connection = TestHelper.testConnection();
+
+        initializeConnectorTestFramework();
+        Testing.Files.delete(TestHelper.DB_HISTORY_PATH);
+    }
+
+    @After
+    public void after() throws SQLException {
+        if (connection != null) {
+            connection.close();
+        }
+    }
+
+    @Test
+    public void caseInsensitiveDatabase() throws Exception {
+        connection.execute(
+                "CREATE TABLE MyTableOne (Id int primary key, ColA varchar(30))",
+                "INSERT INTO MyTableOne VALUES(1, 'a')"
+        );
+        TestHelper.enableTableCdc(connection, "MyTableOne");
+        testDatabase();
+    }
+
+    @Test
+    public void caseSensitiveDatabase() throws Exception {
+        connection.execute(
+                "ALTER DATABASE testDB COLLATE Latin1_General_BIN",
+                "CREATE TABLE MyTableOne (Id int primary key, ColA varchar(30))",
+                "INSERT INTO MyTableOne VALUES(1, 'a')"
+        );
+        TestHelper.enableTableCdc(connection, "MyTableOne");
+        testDatabase();
+    }
+
+    private void testDatabase() throws Exception {
+        final Configuration config = TestHelper.defaultConfig()
+                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL)
+                .build();
+
+        start(SqlServerConnector.class, config);
+        assertConnectorIsRunning();
+
+        SourceRecords records = consumeRecordsByTopic(1);
+        Assertions.assertThat(records.recordsForTopic("server1.dbo.MyTableOne")).hasSize(1);
+        SourceRecord record = records.recordsForTopic("server1.dbo.MyTableOne").get(0);
+        assertSchemaMatchesStruct(
+            (Struct)((Struct)record.value()).get("after"),
+            SchemaBuilder.struct()
+                .optional()
+                .name("server1.testDB.dbo.MyTableOne.Value")
```

Comment:
Not directly related to this PR, but this schema name makes me wonder. What is that "dbo" part in there? Why isn't it "server1.testDB.MyTableOne.Value", i.e. as it'd be for the other connector? In particular, as the "testDB" part is missing in the topic name above. This seems like something we should straighten out?

Reply:
`dbo` is schema name. Connector can connect to only single database, hence I follow the same approach as https://debezium.io/docs/connectors/postgresql/#topic-names so the database name is droppend and replaced with the logical name.

Reply:
> so the database name is droppend and replaced with the logical name.

That makes sense, but it's not what's currently done, is it? I.e. it should be `server1.dbo.MyTableOne.Value` instead of `server1.testDB.dbo.MyTableOne.Value`, or am I missing something?

Reply:
Cool; so btw. that has been my general conclusion on this subject: the DBZ connectors should always only connect to a single database in case a specific DB host can have multiple ones. That database is then represented by the "logical name" in topic and message schema names.

Reply:
This one is a little bit tricky - we are creating envelop schema name from topic name but key/value schema name from table id. Table id is triplet of catalog, schema and table. MySQL has schema set to null, PostgreSQL and Oracle has catalog set to null but SQL Server is really hard-working guy and sets both catalog and and schema which leads to that unusual schema name.

Reply:
@gunnarmorling Updated, now depends on https://github.com/debezium/debezium/pull/739

Reply:
> SQL Server is really hard-working guy

😄 

> Updated

Nice, thanks!